Bosworth Toller's Anglo-Saxon Dictionary Online.js

```diff
@@ -0,0 +1,636 @@
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll(".btd--search-entry");
+	for (const row of rows) {
+		// Don't retrieve the "similar entry" links.
+		const href = attr(row, ".btd--search-entry-header a", "href");
+		let title = ZU.trimInternal(text(row, ".btd--entry-grammar").trim());
+		if (!title) {
+			title = text(row, ".btd--search-entry-header a"); // fallback
+		}
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found && items;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) === 'multiple') {
+		const items = await Z.selectItems(getSearchResults(doc));
+		if (!items) return;
+		for (const url of Object.keys(items)) {
+			scrape(await requestDocument(url));
+		}
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+const BOSWORTH_TOLLER_INFO = {
+	dictionaryTitle: "An Anglo-Saxon Dictionary Online",
+	language: "en",
+	place: "Prague",
+	publisher: "Faculty of Arts, Charles University",
+	date: "2014",
+	originalPublisher: "Oxford University Press",
+	originalPlace: "London",
+	creators: [ZU.cleanAuthor("Joseph Bosworth", "author"),
+		ZU.cleanAuthor("Thomas Northcote Toller", "editor"),
+		ZU.cleanAuthor("Christ Sean", "editor"),
+		ZU.cleanAuthor("Ondřej Tichy", "editor")],
+};
+
+function scrape(doc, url = doc.location.href) {
+	const itemType = detectWeb(doc);
+	if (!itemType || itemType !== "dictionaryEntry") {
+		Z.debug(`Error: item type mismatch for ${url}: ${itemType}`);
+		return;
+	}
+	const item = new Z.Item(itemType);
+
+	// "Constant" fields
+	Object.assign(item, BOSWORTH_TOLLER_INFO);
+	item.rights = ZU.trimInternal(text(doc, ".btd--copyright-citation p")); // first paragraph
+
+	// Page-specific data
+	item.url = url;
+
+	// Word entry
+	item.title = normalizeLemma(doc) || "[Unknown entry]";
+
+	// Original publication and page number in it, if any
+	let tmp = getPage(doc); // may be null, which is OK for assign
+	Object.assign(item, tmp);
+
+	// Notes (summary of word definitions)
+	if ((tmp = getNotes(doc, item.title))) { // NOTE: assignment
+		item.notes = tmp;
+	}
+
+	// Snapshot
+	item.attachments = [{
+		document: doc,
+		title: "Snapshot",
+		mimeType: "text/html"
+	}];
+
+	item.complete();
+}
+
+// See https://bosworthtoller.com/images-dictionary/frontback_matter.pdf
+const BOOK_ORIG_INFO = {
+	b: {
+		originalDictionaryTitle: "An Anglo-Saxon Dictionary, Based on the Manuscript Collections of the Late Joseph Bosworth, D.D., F.R.S.",
+		originalDate: "1898",
+	},
+	d: {
+		originalDictionaryTitle: "An Anglo-Saxon Dictionary, Based on the Manuscript Collections of the Late Joseph Bosworth; Supplement",
+		originalDate: "1921",
+	},
+};
+
+// Get the original volume info and page number by parsing the URL of the
+// scanned page linked to the article. Returns an object with the fields
+// suitably populated.
+function getPage(doc) {
+	const imageURL = attr(doc, ".btd--image-pin-pan > img", "src");
```

Comment:
I believe dstillman commented to this elsewhere: currently Zotero style is to use `const` sparingly -- it doesn't makea difference here (at all, I believe), but if you recall his comment, would be good to apply here (as in: in this translator) as well.